DalamudPlugin/RaidsRewritten/Scripts/Conditions/Temperature.cs

```diff
@@ -0,0 +1,182 @@
+﻿using System;
+using Flecs.NET.Core;
+using RaidsRewritten.Extensions;
+using RaidsRewritten.Game;
+using RaidsRewritten.Log;
+using RaidsRewritten.Scripts.Attacks.Components;
+
+namespace RaidsRewritten.Scripts.Conditions;
+
+public class Temperature(DalamudServices dalamud, ILogger logger) : ISystem
+{
+    private readonly ILogger logger = logger;
+    private Query<Temperature.Component> ComponentQuery;
+    public const string GaugeXPositionConfig = "UCOB Rewritten.TemperatureControlX";
+    public const string GaugeYPositionConfig = "UCOB Rewritten.TemperatureControlY";
+    public const string GaugeImagePath = "temperature_gauge.png";
+
+    public record struct Component()
+    {
+        public float CurrentTemperature = 0.0f;
+        public readonly float MaxTemp = 200.0f;
+        public readonly float MinTemp = -100.0f;
+        public readonly float OverheatTemp = 100.0f;
+        public readonly float DeepfreezeTemp = -100.0f;
+    }
+    public record struct Overheat();
+    public record struct Deepfreeze();
+    public record struct Id(int Value);
+    public record struct TemperatureDelta(float Value, float ReapplicationCooldownTime, bool SelfDestructable = false, bool Applied = false);
+    public static void SetTemperature(Entity playerEntity)
+    {
+        var world = playerEntity.CsWorld();
+        using Query<Temperature.Component> q = playerEntity.CsWorld().QueryBuilder<Temperature.Component>().With(Ecs.ChildOf, playerEntity).Build();
+        if (q.IsTrue()) { return; }
+
+        playerEntity.CsWorld().Entity()
+                .Set(new Temperature.Component())
+                .Set(new Condition.Component("", 9999.0f))
```

Comment:
This reminds me I gotta add no-timer condition support, but I'd do this for now
```suggestion
                .Set(new Condition.Component("", float.PositiveInfinity)
```